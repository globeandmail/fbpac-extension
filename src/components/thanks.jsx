import React from "react";
import { connect } from "react-redux";
import { withI18n } from "i18n.js";
import { Toggler } from "components/toggler.jsx";

const ThanksUnconnected = withI18n(({ getMessage, thanks, ratings_count }) => (
  <div id="thankscontainer">
    <div id="thanksbar">
      {thanks && ratings_count
        ? getMessage(thanks, { count: ratings_count || 0 })
        : null}
    </div>
    <Toggler />
  </div>
));
export const Thanks = connect(state => state)(ThanksUnconnected);
