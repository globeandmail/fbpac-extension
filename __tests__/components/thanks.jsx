import React from "react";
import { mount } from "enzyme";
import configureMockStore from "redux-mock-store";
import { Provider } from "react-redux";
import { Thanks } from "../../src/components/thanks.jsx";

const mockStore = configureMockStore();

describe("thanks", () => {
  it("should render", () => {
    const store = mockStore({
      language: { country: "US", language: "en" },
      ads: [],
      ratings: [],
      thanks: "thanks1"
    });
    const thanks = mount(
      <Provider store={store}>
        <Thanks />
      </Provider>
    );
    expect(thanks).toMatchSnapshot();
  });
});
