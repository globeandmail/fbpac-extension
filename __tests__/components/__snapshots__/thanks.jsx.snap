// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`thanks should render 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Connect(Component))>
    <Connect(Component)
      ads={Array []}
      dispatch={[Function]}
      language={
        Object {
          "country": "US",
          "language": "en",
        }
      }
      ratings={Array []}
      thanks="thanks1"
    >
      <Component
        ads={Array []}
        dispatch={[Function]}
        getMessage={[Function]}
        language={
          Object {
            "country": "US",
            "language": "en",
          }
        }
        ratings={Array []}
        thanks="thanks1"
      >
        <div
          id="thankscontainer"
        >
          <div
            id="thanksbar"
          >
            Thanks! ðŸ™‚
          </div>
          <Connect(TogglerUnconnected)>
            <TogglerUnconnected
              ads={Array []}
              dispatch={[Function]}
              language={
                Object {
                  "country": "US",
                  "language": "en",
                }
              }
              ratings={Array []}
              thanks="thanks1"
            >
              <div
                id="toggler"
              >
                <div
                  id="tabs"
                >
                  <Connect(Component)
                    amount={0}
                    message="rate_ads"
                    type="rater"
                  >
                    <Component
                      amount={0}
                      dispatch={[Function]}
                      getMessage={[Function]}
                      language={
                        Object {
                          "country": "US",
                          "language": "en",
                        }
                      }
                      message="rate_ads"
                      type="rater"
                    >
                      <div
                        className="toggle"
                        onClick={[Function]}
                      >
                        Ads I'm Seeing
                      </div>
                    </Component>
                  </Connect(Component)>
                  <Connect(Component)
                    amount={0}
                    message="see_ads"
                    type="ads"
                  >
                    <Component
                      amount={0}
                      dispatch={[Function]}
                      getMessage={[Function]}
                      language={
                        Object {
                          "country": "US",
                          "language": "en",
                        }
                      }
                      message="see_ads"
                      type="ads"
                    >
                      <div
                        className="toggle"
                        onClick={[Function]}
                      >
                        Ads Others are Seeing
                      </div>
                    </Component>
                  </Connect(Component)>
                </div>
                <div
                  id="container"
                >
                  <Connect(Ratings)
                    ratings={Array []}
                  >
                    <Ratings
                      onRatingClick={[Function]}
                      ratings={Array []}
                    >
                      <div
                        id="ratings"
                      />
                    </Ratings>
                  </Connect(Ratings)>
                </div>
              </div>
            </TogglerUnconnected>
          </Connect(TogglerUnconnected)>
        </div>
      </Component>
    </Connect(Component)>
  </Connect(Connect(Component))>
</Provider>
`;
